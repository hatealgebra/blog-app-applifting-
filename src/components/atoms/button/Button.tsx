import React from 'react';
import StyledButton from './button.styled';
import { ButtonProps } from './button.types';

const Button = ({
  variant = 'standard',
  size = 'md',
  colortheme = 'primary',
  children,
  isBlock,
  type = 'button',
  onClick,
}: ButtonProps) => (
  <StyledButton
    onClick={onClick}
    variant={variant}
    size={size}
    colortheme={colortheme}
    isBlock={isBlock}
    type={type}
  >
    {children}
  </StyledButton>
);

export default Button;
