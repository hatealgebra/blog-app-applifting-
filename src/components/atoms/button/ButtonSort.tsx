import React from 'react';

import { FaSort } from '@react-icons/all-files/fa/FaSort';
import { StyledButtonSort } from './button.styled';

const ButtonSort = ({
  isActive,
  children,
  onClick,
  className,
}: ButtonSortProps) => {
  return (
    <StyledButtonSort
      className={className}
      isActive={isActive}
      onClick={onClick}
      data-testid="button-sort-title"
    >
      {children}
      <FaSort />
    </StyledButtonSort>
  );
};

interface ButtonSortProps {
  isActive?: boolean;
  children: string;
  onClick: () => void;
  className?: string;
}

export default ButtonSort;
