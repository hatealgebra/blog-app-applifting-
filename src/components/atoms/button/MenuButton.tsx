import React from "react";

import { CgMenu } from "@react-icons/all-files/cg/CgMenu";
import { MenuButtonContainer } from "./button.styled";

const MenuButton = ({ onClick }: MenuButtonProps) => {
  return (
    <MenuButtonContainer onClick={onClick}>
      <CgMenu size="20px" />
    </MenuButtonContainer>
  );
};

export interface MenuButtonProps {
  onClick: () => void;
}

export default MenuButton;
