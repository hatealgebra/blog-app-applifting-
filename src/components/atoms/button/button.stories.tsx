import React from 'react';
import { action } from '@storybook/addon-actions';
import { ComponentMeta } from '@storybook/react';
import Button from './Button';
import MenuButton from './MenuButton';
import ButtonSort from './ButtonSort';

export const StandardButtons = () => (
  <>
    <Button colortheme="primary" onClick={() => action('Button clicked')}>
      Primary
    </Button>
    <Button onClick={() => action('Button clicked')} colortheme="secondary">
      Secondary
    </Button>
  </>
);

export const OutlineButtons = () => (
  <>
    <Button variant="outline" onClick={() => action('Button clicked')}>
      Primary
    </Button>
    <Button
      variant="outline"
      colortheme="secondary"
      onClick={() => action('Button clicked')}
    >
      Secondary
    </Button>
  </>
);

export const LargeStandardButtons = () => (
  <>
    <Button
      size="lg"
      colortheme="primary"
      onClick={() => action('Button clicked')}
    >
      Primary
    </Button>
    <Button
      size="lg"
      onClick={() => action('Button clicked')}
      colortheme="secondary"
    >
      Secondary
    </Button>
  </>
);
export const LargeOutlineButtons = () => (
  <>
    <Button
      size="lg"
      colortheme="primary"
      variant="outline"
      onClick={() => action('Button clicked')}
    >
      Primary
    </Button>
    <Button
      size="lg"
      onClick={() => action('Button clicked')}
      colortheme="secondary"
      variant="outline"
    >
      Secondary
    </Button>
  </>
);
export const SmallStandardButtons = () => (
  <>
    <Button
      size="sm"
      colortheme="primary"
      onClick={() => action('Button clicked')}
    >
      Primary
    </Button>
    <Button
      size="sm"
      onClick={() => action('Button clicked')}
      colortheme="secondary"
    >
      Secondary
    </Button>
  </>
);
export const SmallOutlineButtons = () => (
  <>
    <Button
      size="sm"
      colortheme="primary"
      variant="outline"
      onClick={() => action('Button clicked')}
    >
      Primary
    </Button>
    <Button
      size="sm"
      onClick={() => action('Button clicked')}
      colortheme="secondary"
      variant="outline"
    >
      Secondary
    </Button>
  </>
);
export const BlockButton = () => (
  <div style={{ maxWidth: '200px', width: '100%' }}>
    <Button onClick={() => action('Button clicked!')} isBlock>
      Block Button
    </Button>
  </div>
);

export const MenuButtonExample = () => (
  <MenuButton onClick={() => action('Menu button clicked!')} />
);

export const SortButtonAllVariants = () => (
  <>
    <ButtonSort dispatchSort={action('Clicked')}>Article Title</ButtonSort>
    <ButtonSort dispatchSort={action('Clicked')} isActive>
      Article Title Active
    </ButtonSort>
  </>
);

export default {
  component: Button,
  subcomponents: { MenuButtonExample },
  title: 'Atoms/Button',
  decorators: [
    (Story) => (
      <div style={{ margin: '1em', display: 'flex', gap: '20px' }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Button | typeof MenuButtonExample>;
