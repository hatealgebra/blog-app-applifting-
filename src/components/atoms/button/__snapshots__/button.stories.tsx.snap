// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button BlockButton smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <div style="max-width: 200px; width: 100%;">
    <button variant="standard"
            colortheme="primary"
            type="button"
            class="sc-hLclGa erOOOs"
    >
      Block Button
    </button>
  </div>
</div>
`;

exports[`Atoms/Button LargeOutlineButtons smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button variant="outline"
          colortheme="primary"
          type="button"
          class="sc-hLclGa eeiCuf"
  >
    Primary
  </button>
  <button variant="outline"
          colortheme="secondary"
          type="button"
          class="sc-hLclGa eHpWOv"
  >
    Secondary
  </button>
</div>
`;

exports[`Atoms/Button LargeStandardButtons smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button variant="standard"
          colortheme="primary"
          type="button"
          class="sc-hLclGa fiXdBi"
  >
    Primary
  </button>
  <button variant="standard"
          colortheme="secondary"
          type="button"
          class="sc-hLclGa jWKvye"
  >
    Secondary
  </button>
</div>
`;

exports[`Atoms/Button MenuButtonExample smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button class="sc-dkjKgF iiHxSZ">
    <svg stroke="currentColor"
         fill="none"
         stroke-width="0"
         viewbox="0 0 24 24"
         height="20px"
         width="20px"
         xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M2 6C2 5.44772 2.44772 5 3 5H21C21.5523 5 22 5.44772 22 6C22 6.55228 21.5523 7 21 7H3C2.44772 7 2 6.55228 2 6Z"
            fill="currentColor"
      >
      </path>
      <path d="M2 12.0322C2 11.4799 2.44772 11.0322 3 11.0322H21C21.5523 11.0322 22 11.4799 22 12.0322C22 12.5845 21.5523 13.0322 21 13.0322H3C2.44772 13.0322 2 12.5845 2 12.0322Z"
            fill="currentColor"
      >
      </path>
      <path d="M3 17.0645C2.44772 17.0645 2 17.5122 2 18.0645C2 18.6167 2.44772 19.0645 3 19.0645H21C21.5523 19.0645 22 18.6167 22 18.0645C22 17.5122 21.5523 17.0645 21 17.0645H3Z"
            fill="currentColor"
      >
      </path>
    </svg>
  </button>
</div>
`;

exports[`Atoms/Button OutlineButtons smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button variant="outline"
          colortheme="primary"
          type="button"
          class="sc-hLclGa kMBCXK"
  >
    Primary
  </button>
  <button variant="outline"
          colortheme="secondary"
          type="button"
          class="sc-hLclGa cYUcwG"
  >
    Secondary
  </button>
</div>
`;

exports[`Atoms/Button SmallOutlineButtons smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button variant="outline"
          colortheme="primary"
          type="button"
          class="sc-hLclGa keduQX"
  >
    Primary
  </button>
  <button variant="outline"
          colortheme="secondary"
          type="button"
          class="sc-hLclGa eNYRZn"
  >
    Secondary
  </button>
</div>
`;

exports[`Atoms/Button SmallStandardButtons smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button variant="standard"
          colortheme="primary"
          type="button"
          class="sc-hLclGa dZkWcq"
  >
    Primary
  </button>
  <button variant="standard"
          colortheme="secondary"
          type="button"
          class="sc-hLclGa bLrldS"
  >
    Secondary
  </button>
</div>
`;

exports[`Atoms/Button SortButtonAllVariants smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button class="sc-grXZZQ kImwcT">
    <span>
      Article Title
    </span>
    <svg stroke="currentColor"
         fill="currentColor"
         stroke-width="0"
         viewbox="0 0 320 512"
         height="1em"
         width="1em"
         xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
      </path>
    </svg>
  </button>
  <button class="sc-grXZZQ gggWLR">
    <span>
      Article Title Active
    </span>
    <svg stroke="currentColor"
         fill="currentColor"
         stroke-width="0"
         viewbox="0 0 320 512"
         height="1em"
         width="1em"
         xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Atoms/Button StandardButtons smoke-test 1`] = `
<div style="margin: 1em; display: flex; gap: 20px;">
  <button variant="standard"
          colortheme="primary"
          type="button"
          class="sc-hLclGa dhTwVn"
  >
    Primary
  </button>
  <button variant="standard"
          colortheme="secondary"
          type="button"
          class="sc-hLclGa foIDLL"
  >
    Secondary
  </button>
</div>
`;
